src/pages/auth.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {useState} from 'react';
+import {NextPage} from 'next';
+import {FaDiscord, FaLock, FaGithub, FaGoogle} from 'react-icons/fa';
+import {NextAppPageProps} from '../types/app';
+import Layout from '../components/Layout';
+import Spinner from '../components/Spinner';
+import {useFormFields} from '../lib/utils';
+import {useAuth} from '../lib/auth';
+
+type SignUpFieldProps = {
+  email: string;
+  password: string;
+};
+
+const FORM_VALUES: SignUpFieldProps = {
+  email: '',
+  password: '',
+};
+
+const AuthPage: NextPage<NextAppPageProps> = () => {
+  const [isSignIn, setIsSignIn] = useState(true);
+  const {loading, signIn, signUp, signInWithProvider} = useAuth();
+  // Now since we have our form ready, what we're gonna need for signing up our users
+  // 1. let users provide email and password
+  const [values, handleChange, resetFormFields] =
+    useFormFields<SignUpFieldProps>(FORM_VALUES);
+  // 2. send the provided details to Supabase
+
+  const handleSubmit = (event: React.FormEvent) => {
```

Comment:
white space for comment
```suggestion

  // 2. send the provided details to Supabase
  const handleSubmit = (event: React.FormEvent) => {
```